lib/config.js

```diff
@@ -229,6 +230,16 @@ function getLocalConfig(thisConfig, directory) {
             continue;
         }
 
+        // Check for root flag
+        if (localConfig.root === true) {
+            rootPath = path.dirname(localConfigFile);
+        }
+
+        // If root flag is set, don't consider file if it is above root
+        if (rootPath && localConfigFile.indexOf(rootPath) === -1) {
```

Comment:
Please use https://github.com/domenic/path-is-inside module to figure out if a particular path is inside another path.
Certain times your condition would fail. This module talks about those conditions.